test/Function/StackArgsWithFormals.js

```diff
@@ -0,0 +1,168 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+//Test1 : Tests Optimization with bailout.
+var shouldBailout = false;
+var test1 = function (argMath3) 
+{
+    if(shouldBailout)
+    {
+        arguments[0] = 4;
+        for(var o in arguments)
+        {
+            print(o);
+        }
+    }
+    WScript.Echo(argMath3);

```

Comment:
Done.

---
In reply to: [67927884](https://github.com/Microsoft/ChakraCore/pull/1135#discussion_r67927884) [](ancestors = 67927884)